modules/standard/GPU.chpl

```diff
@@ -224,4 +225,133 @@ module GPU
 
     chpl_gpu_set_peer_access(loc1Sid, loc2Sid, shouldEnable);
   }
+
+  // ============================
+  // Atomics
+  // ============================
+
+  // In the runtime library we have various type specific wrappers to call out
+  // to the CUDA/ROCM atomic operation functions.  Note that the various
+  // CUDA/ROCM atomic functions are defined in terms of the various "minimum
+  // width" C types (like int, long, etc.) rather than fixed width types (like
+  // int32_t, int64_t, etc.) thus we need to figure out which of these C types
+  // makes the "best fit" for a corresponding Chapel type.
+  private proc atomicExternTString(type T) param {
+    param nb = if isNumeric(T) then numBits(T) else -1;
+    param nbInt = numBits(c_int);
+    param nbShort = numBits(c_short);
+    param nbFloat = numBits(c_float);
+
+    if nb == -1 then return "unknown";
+    if isUint(T) && nb <= nbShort then return "short";
+    if isInt(T)  && nb <= nbInt   then return "int";
+    if isInt(T)                   then return "longlong";
+    if isUint(T) && nb <= nbInt   then return "uint";
+    if isUint(T)                  then return "ulonglong";
+    if isReal(T) && nb <= nbFloat then return "float";
+    if isReal(T)                  then return "double";
+    return "unknown";
+  }
+
+  private proc externFunc(param opName : string, type T) param {
+    return "chpl_gpu_atomic_" + opName + "_" + atomicExternTString(T);
+  }
+
+  // used to indicate that although a given atomic operation
+  // is supported by other SDKs these particular ones are not
+  // supported by ROCm.
+  private proc invalidGpuAtomicOpForRocm(param s : string) param {
+    select s { when
+      "chpl_gpu_atomic_min_longlong",
+      "chpl_gpu_atomic_max_longlong"
```

Comment:
I didn't know you could have a comma-separated list of values in `when` clause. Chapel is a good language.